### ISSUE TITLE
Flutter inspector does not load for Flutter web apps

### ISSUE THREAD

**Issue by WangMiaoGit** [flutter/devtools#3105]
**In VScode，Widget Inspector show like this picture.**
![image](https://user-images.githubusercontent.com/22518556/121505347-525eca80-ca15-11eb-80f1-6bcfc9fc771e.png)

it also on web browser
![image](https://user-images.githubusercontent.com/22518556/121505376-5b4f9c00-ca15-11eb-8f04-6d9424bb6a9b.png)

**Comment by DanTup** [comment]
I can reproduce this, and it seems to only occur when running the Flutter app for the web. If I launch from VS Code, navigating to the DevTools web app manually and pasting in the VM Service seems to work, however from the command line it seems to fail in this way always.

- Checkout Flutter v2.2.1
- Run `flutter run -d chrome`
- Run `pub global run devtools`
- Paste the VM Service URI from the `flutter` output into the connection dialog

<img width="1465" alt="Screenshot 2021-06-10 at 11 14 55" src="https://user-images.githubusercontent.com/1078012/121508045-1c9def80-c9dd-11eb-9735-c7f7b6af2ebe.png">

**Comment by WangMiaoGit** [comment]
I thought it was the VS code version, but my colleague's Android Studio also had this problem.

**Comment by DanTup** [comment]
@jacob314 this looks like a race condition, though I'm not familiar enough to know how this should work. I added some prints to `initState` in `InspectorScreenBodyState`:

```dart
print('In InspectorScreenBodyState.initState, subscribing to onConnectionAvailable!');
autoDispose(serviceManager.onConnectionAvailable.listen(_handleConnectionStart));

if (serviceManager.connectedAppInitialized) {
  print('connectedAppInitialized=true so immediately calling _handleConnectionStart');
  _handleConnectionStart(serviceManager.service);
}
```

It looks like this code is intended to call `_handleConnectionStart` immediately if there's already a connection, but otherwise call it when it becomes available. However I also logged when the `onConnectionAvailable` stream fires, and it appears to occur _before_ this code runs, but `connectedAppInitialized` is still `false`, so neither of these fire:

![Screenshot 2021-06-10 at 12 05 00](https://user-images.githubusercontent.com/1078012/121514598-1cedb900-c9e4-11eb-9cdb-0176b98a2833.png)

Since the log shows `onConnectionAvailable` was already triggered, I'd have expected `'connectedAppInitialized=true so immediately calling _handleConnectionStart'` to also be printed.
